hoprd/keypair/src/key_pair.rs

```diff
@@ -61,11 +59,33 @@ impl Aes128Ctr {
     }
 }
 
-pub struct IdentityOptions {
-    pub initialize: bool,
-    pub id_path: String,
-    pub password: String,
-    pub private_key: Option<Box<[u8]>>,
+pub enum IdentityRetrievalModes<'a> {
+    /// hoprd starts with a previously generated identitiy file.
+    /// If none is present, create a new one
+    FromFile {
+        /// Used encrypt / decrypt identity file
+        password: &'a str,
+        /// path to store / load identity file
+        id_path: &'a str,
+    },
+    /// hoprd receives at startup a private which it will use
+    /// for the entire runtime. The private stays in memory and
+    /// thus remains fluent
+    FromPrivateKey {
+        /// hex string containing the private key
+        private_key: &'a str,
+    },
+    /// takes a private key and create an identity file at the
+    /// provided file destination
+    #[cfg(any(advanced, test))]
```

Comment:
This does not look right, it should be something like `#[cfg(any(feature = "hopli", test))]` and then the `hopli` feature needs to be defined in Cargo.toml

Reply:
good point, changed